From c904a0c79d034556b0f1fb08cedbb773f877ca2b Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Fri, 19 Jan 2024 09:14:09 +0530
Subject: [PATCH 79/95] arm64: dts: st: Add wakeup source for usbh using exti
 on stm32mp25

Remote-wakeup capable device connected to USBH can act as a
source to wakeup system from low-power state on mp25 platform
Enable exti-intr connected to utmi_suspend_o_n output of USB2H
as wakeup-source

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/linux.git f8ac94525280f933cb600fa8f26cef6091389bcd arm64: dts: st: Add wakeup source for usbh using exti on stm32mp25]
Change-Id: Id85ab9b08149327fa9a1409d1d081465628282da
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/370803
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index e14d888ed27ef..436a887fb967d 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -1787,6 +1787,8 @@ usbh: usb@482e0000 {
 				#size-cells = <1>;
 				ranges = <0x482e0000 0x482e0000 0x20000>;
 				access-controllers = <&rifsc 63>;
+				wakeup-source;
+				interrupts-extended = <&exti1 43 IRQ_TYPE_EDGE_RISING>;
 				status = "disabled";
 
 				usbh_ohci: usb@482e0000 {
@@ -1797,6 +1799,7 @@ usbh_ohci: usb@482e0000 {
 					interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
 					phys = <&usb2_phy1>;
 					phy-names = "usb";
+					wakeup-source;
 				};
 
 				usbh_ehci: usb@482f0000 {
@@ -1808,6 +1811,7 @@ usbh_ehci: usb@482f0000 {
 					companion = <&usbh_ohci>;
 					phys = <&usb2_phy1>;
 					phy-names = "usb";
+					wakeup-source;
 				};
 			};
 
-- 
2.51.0

